### PR TITLE
Add CAN FD support to canutils reader and writer

### PR DESCRIPTION
**Pull request by simontegelid**
This PR adds CAN FD support to the canutils reader and writer. I also sneaked in a small formatting improvement to use bytes.hex instead of the str/list/join-thing.

### CODE REVIEW

File: test/data/example_data.py [hunk 1]
```diff
@@ -112,8 +112,6 @@ def sort_messages(messages):
     [
         Message(is_fd=True, data=range(64)),
         Message(is_fd=True, data=range(8)),
-        Message(is_fd=True, bitrate_switch=True, is_remote_frame=True),
```
Comment (felixdivo) [inline test/data/example_data.py:115]: Why did you remove these?

**Comment by felixdivo** [comment]
Nice addition!

File: test/data/example_data.py (hunk 1 above)
Comment (simontegelid) [reply test/data/example_data.py:115]: I explained it in the commit message: "Remove RTR tests for FD frames since they doesn't exist
and thus cannot be represented in the canutils log format.
See "CAN with Flexible Data-Rate, version 1.0, Bosch,
April 17th, 2012, pp.11": "The REMOTE TRANSMISSION REQUEST
(RTR) bit only exists in CAN format frames. [...] There are
no REMOTE FRAMES in CAN FD format." "

Comment (felixdivo) [reply test/data/example_data.py:115]: Ah, nice catch! Sorry, I did not recognize the additional lines in the message. Looks fine.

**Comment by felixdivo** [comment]
Could you run `black python-can/can/io/canutils.py` once to fix the formatting complaint [here](https://github.com/hardbyte/python-can/pull/1042/checks?check_run_id=2534489978#step:5:73)? Else, this looks fine to be merged.

**Review by felixdivo — APPROVED** [review]

**Comment by simontegelid** [comment]
I think it should be fine now!

**Review by hardbyte — APPROVED** [review]
Thanks!
